src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  constructor() {
+    super();
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor() {
```

Comment:
[FUNCTION NAMING] - The constructor of `Carnivore` should accept a `name` parameter to be passed to the `super` constructor. Without this, all `Carnivore` instances would be created without a name, which violates the task requirement that all animals must have a `name` property.